sui-execution/latest/sui-move-natives/src/eth_state_proof.rs

```diff
@@ -73,12 +222,36 @@ pub(crate) fn create_initial_eth_state_data(
     _ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    todo!();
-}
+    // Load the cost parameters from the protocol config
+    let verify_message_proof_cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .eth_state_proof
+        .clone();
+
+    // Charge the base cost for this operation.
+    native_charge_gas_early_exit!(
+        context,
+        verify_message_proof_cost_params.create_initial_eth_state_data_cost_base
+    );
+
+    let cost = context.gas_used();
+
+    let network = pop_arg!(args, Vector).to_vec_u8()?;
+
+    let network = String::from_utf8(network)
+        .map_err(|_| PartialVMError::new(StatusCode::FAILED_TO_DESERIALIZE_ARGUMENT))?;
+
+    let file_path = format!("crates/eth-initial-state/eth_state_{}.bcs", network);
+
+    let mut file = File::open(file_path)
+        .map_err(|_| PartialVMError::new(StatusCode::VALUE_SERIALIZATION_ERROR))?;
+    let mut eth_state_bytes = Vec::new();
+    File::read_to_end(&mut file, &mut eth_state_bytes)
+        .map_err(|_| PartialVMError::new(StatusCode::VALUE_SERIALIZATION_ERROR))?;
 
```

Comment:
Wait wait, I'm really uncertain that we can do this from the validator side. You are reading a file from the file system here? it's not common to do I/O like this in a native function. Let's think where else we can put this. One idea is indeed to do this in Genesis. If this is the plan, let's document this with a `TODO` here, and in an issue.

Reply:
As we talked with Omer yesterday, there is no plan to insert this into genesis process.
Do you have any idea how else we can do this?
This is a constant data that we need to have hardcoded somehow, and I can't figure out another way to do this (this is huge data that we can't just hardcode into the .rs file)

Reply:
@omersadika